File: Assignment 2/part2_memory_management/memory_management.cpp
#include <iostream>
using namespace std;

int main() {
    int* arr = new int[5]{1, 2, 3, 4, 5};
    int sum = 0;
    for (int i = 0; i < 5; ++i) {
        sum += arr[i];
    }
    cout << "Sum: " << sum << endl;
    delete[] arr;
    return 0;
}

File: Assignment 2/part2_memory_management/memory_management.java
public class MemoryDemo {
    public static void main(String[] args) {
        int[] data = new int[]{1, 2, 3, 4, 5};
        System.out.println("Sum: " + calculateSum(data));
    }

    public static int calculateSum(int[] arr) {
        int total = 0;
        for (int num : arr) {
            total += num;
        }
        return total;
    }
}
